backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardlist/stack/internal/BpkStackCardListImpl.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.button.BpkButtonIconPosition
+import net.skyscanner.backpack.compose.button.BpkButtonType
+import net.skyscanner.backpack.compose.cardlist.stack.BpkStackCardAccessoryStyle
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.sectionheader.BpkSectionHeader
+import net.skyscanner.backpack.compose.sectionheader.BpkSectionHeaderButton
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.ChevronDown
+import net.skyscanner.backpack.compose.tokens.ChevronUp
+
+@Composable
+internal fun BpkStackCardListImpl(
+    title: String,
+    description: String?,
+    totalCount: Int,
+    modifier: Modifier = Modifier,
+    accessoryStyle: BpkStackCardAccessoryStyle? = null,
+    headerButton: BpkSectionHeaderButton? = null,
+    accessibilityHeaderTagEnabled: Boolean? = true,
+    content: @Composable ((Int) -> Unit),
+) {
+
+    var isExpanded by rememberSaveable { mutableStateOf(false) }
+    val targetCount = when (accessoryStyle) {
+        is BpkStackCardAccessoryStyle.Expand -> {
+            if (accessoryStyle.expandedCount < 0 || accessoryStyle.collapsedCount < 0) {
+                throw IllegalArgumentException("expandedCount/collapsedCount can not be negative")
+            }
+            if (accessoryStyle.expandedCount < accessoryStyle.collapsedCount) {
+                throw IllegalArgumentException("expandedCount should be greater than collapsedCount")
+            }
+            if (accessoryStyle.expandedCount <= MINIMUM_EXPANDED_COUNT) {
+                totalCount
+            } else {
+                if (isExpanded) accessoryStyle.expandedCount else accessoryStyle.collapsedCount
+            }
+        }
+        else -> totalCount
+    }
+
+    LazyColumn(
+        modifier = modifier,
+        contentPadding = PaddingValues(start = BpkSpacing.Base, end = BpkSpacing.Base),
+        verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+    ) {
+        item {
+            BpkSectionHeader(
+                title = title,
+                description = description,
+                modifier = Modifier,
+                button = headerButton,
+                accessibilityHeaderTagEnabled = accessibilityHeaderTagEnabled,
+            )
+        }
+
+        items(targetCount) {
+            Box(Modifier.animateItem()) {
+                content(it)
+            }
+        }
+
+        item {
+            accessoryStyle?.let {
+                when (it) {
+                    is BpkStackCardAccessoryStyle.Expand -> {
+                        if (it.expandedCount > MINIMUM_EXPANDED_COUNT) {
+                            BpkButton(
+                                modifier = Modifier
```

Comment:
We could add `animateItem()` to the accessory modifier as well, just to make the movement of the expand button smoother. I'd prefer it to slide down out of the way rather than just fade in and out, but maybe @henrik-sky you have a better solution for this?
```suggestion
                                modifier = Modifier
                                    .animateItem()
```

Reply:
I think I have the solution:
```
accessoryStyle?.let {
    when (it) {
        is BpkStackCardAccessoryStyle.Expand -> {
            item(it.expandText) {
                BpkButton(
                    modifier = Modifier
                        .animateItem()
                        .fillMaxWidth()
                        .padding(bottom = BpkSpacing.Base),
                    text = if (isExpanded) it.collapsedText else it.expandText,
                    icon = if (isExpanded) BpkIcon.ChevronUp else BpkIcon.ChevronDown,
                    position = BpkButtonIconPosition.End,
                    type = BpkButtonType.Link,
                    onClick = {
                        isExpanded = !isExpanded
                        it.onExpansionChange?.invoke(isExpanded)
                    },
                )
            }
        }
        is BpkStackCardAccessoryStyle.Button -> {
            if (headerButton == null) {
                item(it.title) {
                    BpkButton(
                        modifier = Modifier
                            .animateItem()
                            .fillMaxWidth()
                            .padding(bottom = BpkSpacing.Base),
                        text = it.title,
                        type = BpkButtonType.Primary,
                        position = BpkButtonIconPosition.End,
                        icon = it.icon,
                        onClick = it.onClick,
                    )
                }
            }
        }
    }
}
```

https://github.com/user-attachments/assets/271a7169-bcc7-4252-82f0-faeb039b9e1f


The individual componsables should be wrapped with `item` and a key provided so the list knows how to animate the position of the item as it knows it is the same item and so should change position not appear/disappear.

Reply:
Ah of course, makes perfect sense now that you explain it 😂  Thanks again for the snippet!

Reply:
Great suggestion @henrik-sky. I applied your suggestion and the accessory buttons animate now!